client/src/pages/RegisterPage/RegisterPage.js

```diff
@@ -16,15 +18,42 @@ const Register = ({ setAlert, register, isAuthenticated, alert, guestMode, toggl
     password2: '',
   })
 
+  const dispatch = useDispatch()
+  const message = useSelector(state => state.auth.message)
+  const error = useSelector(state => state.auth.error)
+
   const { name, email, password, password2 } = formData
 
+  // listen for errors from our server and display them, then reset in
+  useEffect(()=>{
+    if (error) {
+      showToast(error, 'warn')
+      resetAuthMessages(dispatch)
+    }
+  }, [error])
+
+  // listen to messages from our server and display them, then reset it
+  useEffect(()=>{
+    if (message) {
+      showToast(message, 'success')
+      resetAuthMessages(dispatch)
+    }
+  }, [message])
+
   const onChange = (e) => setFormData({ ...formData, [e.target.name]: e.target.value })
 
   const onSubmit = async (e) => {
     e.preventDefault()
-    if (password !== password2) {
-      setAlert('Passwords do not match', 'danger')
-    } else {
+    if (!name || !email || !password || !password2 )
+      showToast('All fields are required', 'warn')
+    else if (password !== password2) 
+      showToast('Passwords do not match', 'warn')
+    else if (!/^[A-Za-z][A-Za-z0-9]*$/.test(name))
+      showToast('Username must contain only letters and numbers', 'warn')
+    else if (!/^\S+@\S+\.\S+$/.test(email))
+      showToast('Please enter a valid email', 'warn')
```

Comment:
All of this _should_ ideally be contained in our reducers as well, though it's a little harder to do without toolkit

Reply:
Ok interesting, yeah I kinda tossed them into our form in the name of checking on the front-end first, I can look into this